java/arcs/android/common/resurrection/ResurrectionRequest.kt

```diff
@@ -167,5 +177,20 @@ data class ResurrectionRequest(
                 notifiers
             )
         }
+
+        /**
+         * Given an [intent] received by the [ResurrectorService] from a client, if it's a request
+         * to unsubscribe from resurrection - gets the component name of the component wishing to
+         * unsubscribe.
+         */
+        @VisibleForTesting(otherwise = VisibleForTesting.PACKAGE_PRIVATE)
+        fun componentNameFromUnrequestIntent(intent: Intent?): ComponentName? {
```

Comment:
For this, and the method above: once the intent's action is determined to be request resurrection or request no resurrection, would it be useful to make some noise if other required fields on the intent for those actions are not set, rather than just silently returning?

Reply:
I could see either way having pros/cons.  The current way could be frustrating for developers who aren't sure why their intent isn't actually being carried out, while the way you're proposing could be frustrating as well - though in a separate fashion (spammy log output). I'll think about this some more and consider changing its behavior in a future PR.